File: Patterns/Pattern 11.cpp
class Solution {
public:
    void pattern11(int n) {
        for (int i = 0; i<n; i++)
        {
            int start = 1;
            if(i%2) start = 0;
            for(int j = 0; j<=i; j++)
            {
                cout << start << " ";
                start = !start;
            }
            cout << endl;

        }

    }
};

File: Patterns/Pattern 11.py
"""
📝 Question:
Print a binary triangle with alternating 1s and 0s.
Odd rows start with 1, even rows start with 0.

For n = 4:
  1
  0 1
  1 0 1
  0 1 0 1
"""

# Pattern 11: Binary Triangle (alternating 1 and 0)
# 1
# 0 1
# 1 0 1
# 0 1 0 1

def pattern11(n):
    for i in range(n):
        start = 1 if i % 2 == 0 else 0
        row = []
        for j in range(i + 1):
            row.append(str(start))
            start = 1 - start
        print(" ".join(row))

pattern11(5)
